src/snapshot/tests/test.rs

```diff
@@ -116,3 +116,115 @@ fn test_hardcoded_snapshot_deserialization() {
 
     assert_eq!(restored_struct, expected_struct);
 }
+
+#[test]
+fn test_invalid_format_version() {
+    #[rustfmt::skip]
+    let invalid_format_snap: &[u8] = &[
+        // This blob is consisted of the following: magic_id (8 bytes),
+        0xAA, 0xAA,
+        #[cfg(target_arch = "aarch64")]
+        0xAA,
+        #[cfg(target_arch = "aarch64")]
+        0xAA,
+        #[cfg(target_arch = "x86_64")]
+        0x64,
+        #[cfg(target_arch = "x86_64")]
+        0x86,
+        0x84, 0x19, 0x10, 0x07,
+        // target version (2 bytes) +
+        0x01, 0x00,
+        // `a` field +
+        0x10, 0x00, 0x00, 0x00,
+        // `b` field: Option variant type (1 byte) + inner enum variant type (4 bytes)
+        // + inner enum value (4 bytes).
+        0x01, 0x01, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00,
+    ];
+
+    let mut snapshot_blob = invalid_format_snap;
+    let mut result: Result<A, Error> = Snapshot::load(&mut snapshot_blob, VersionMap::new());
```

Comment:
You could drop `snapshot_blob` here as well as in the next snapshot loads and just pass the reference to the bytes.

```suggestion
    let mut result: Result<A, Error> = Snapshot::load(&mut invalid_format_snap.as_ref(), VersionMap::new());
```

Reply:
done